File: src/de/boxxit/statis/SerializableException.java
package de.boxxit.statis;

import com.esotericsoftware.kryo.DefaultSerializer;

/**
 * User: Christian Fruth
 */
@DefaultSerializer(SerializableExceptionSerializer.class)
public class SerializableException extends RuntimeException
{
	private String id;

	public SerializableException(String id, String message)
	{
		super(message);
		this.id = id;
	}

	public SerializableException(Throwable ex)
	{
		super(ex.getMessage());
		this.id = ex.getClass().getName();
	}

	public String getId()
	{
		return id;
	}
}

File: stasis-ios/Stasis/SerializableException.m
//
//  SerializableException.m
//  Stasis
//
//  Created by Christian Fruth on 06.03.13.
//  Copyright (c) 2013 Boxx IT Solutions e.K. All rights reserved.
//

#import "SerializableException.h"
#import "SerializableExceptionSerializer.h"

@implementation SerializableException

+ (instancetype)exceptionWithType:(NSString *)type message:(NSString *)message andStackSymbols:(NSArray *)stackSymbols
{
	return [[SerializableException alloc] initWithType:type message:message andStackSymbols:stackSymbols];
}

- (id)initWithType:(NSString *)type message:(NSString *)message andStackSymbols:(NSArray *)stackSymbols
{
	self = [super initWithName:type reason:message userInfo:nil];
	
	if (self != nil)
	{
		_stackSymbols = stackSymbols;
	}
	
	return self;
}

- (NSArray *)callStackSymbols
{
	return _stackSymbols;
}

+ (NSString *)serializingAlias
{
	return @"de.boxxit.statis.SerializableException";
}

+ (Class)defaultSerializer
{
	return [SerializableExceptionSerializer class];
}

@end
